src/main.jsx

```diff
@@ -29,39 +33,67 @@ if (localStorage.VITE_PROJECT_VERSION !== VITE_PROJECT_VERSION) {
   localStorage.clear();
 }
 
-const theme = responsiveFontSizes(
-  createTheme({
-    palette: {
-      primary: {
-        main: '#adb31b',
-        contrastText: '#fff',
-      },
-      secondary: {
-        main: '#f50057',
-      },
-    },
-  }),
-);
+const useThemePrefs = createPersistedState('themePref');
 
 const queryClient = new QueryClient();
 
+function App() {
+  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
+  const [themePrefs, setThemePrefs] = useThemePrefs();
+  const [currentTheme, setCurrentTheme] = useState();
+
+  useEffect(() => {
+    if (themePrefs === "dark") {
+      setCurrentTheme('dark');
+    } else if (themePrefs === 'light') {
+      setCurrentTheme('light');
+    } else {
+      setCurrentTheme(prefersDarkMode ? 'dark' : 'light');
+      setThemePrefs('system');
+    }
```

Comment:
Si on veut que le thème par défaut soit plutôt le "clair" pour ne pas changer les habitudes, il faut remplacer ici : 

```suggestion
    if (themePrefs === "dark") {
      setCurrentTheme('dark');
    } else if (themePrefs === 'light') {
      setCurrentTheme('light');
    } else if (themePrefs === 'system') {
      setCurrentTheme(prefersDarkMode ? 'dark' : 'light');
    } else {
      setThemePrefs('light');
    }
```